ecmascript/transforms/compat/src/es2017/async_to_generator.rs

```diff
@@ -211,6 +213,35 @@ impl Fold for Actual {
                 );
             }
 
+            Expr::Assign(assign_expr) => {
+                // flag if expression is assignment to prototype chain should not try to bind
+                // this context
+                match &assign_expr.left {
+                    PatOrExpr::Pat(pat) => match &**pat {
+                        Pat::Expr(pat_expr) => match &**pat_expr {
+                            Expr::Member(MemberExpr {
+                                obj: ExprOrSuper::Expr(ex),
+                                ..
+                            }) => match &**ex {
+                                Expr::Member(MemberExpr { prop, .. }) => match &**prop {
```

Comment:
Same here, as `a.f[prototype]` is different from `a.f.prototype`.
(I don't think someone to write such code, but it's not correct anyway.